lib/bundler/dsl.rb

```diff
@@ -290,25 +290,11 @@ def add_git_sources
         warn_deprecated_git_source(:github, <<-'RUBY'.strip, 'Change any "reponame" :github sources to "username/reponame".')
 "https://github.com/#{repo_name}.git"
         RUBY
-        # It would be better to use https instead of the git protocol, but this
-        # can break deployment of existing locked bundles when switching between
-        # different versions of Bundler. The change will be made in 2.0, which
-        # does not guarantee compatibility with the 1.x series.
-        #
-        # See https://github.com/bundler/bundler/pull/2569 for discussion
-        #
-        # This can be overridden by adding this code to your Gemfiles:
-        #
-        #   git_source(:github) do |repo_name|
-        #     repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
-        #     "https://github.com/#{repo_name}.git"
-        #   end
         repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
-        # TODO: 2.0 upgrade this setting to the default
-        if Bundler.settings["github.https"]
-          Bundler::SharedHelpers.major_deprecation 3, "The `github.https` setting will be removed"
+        if Bundler.feature_flag.github_https?
           "https://github.com/#{repo_name}.git"
         else
+          Bundler::SharedHelpers.major_deprecation 3, "The `github.https` setting will be removed"
```

Comment:
I think deprecating this setting now needs to be moved to https://github.com/bundler/bundler/blob/66e4215dbf87fedf2fdff3532a0742340c4e02f2/lib/bundler/feature_flag.rb, since here we can't be sure whether we are using the default value for the setting, or whether the user is actually setting it. The deprecation warning needs to be displayed only if the user is actually configuring the setting.

Reply:
Well, I moved this because I happen to know that the default for this setting is now true, so if we end up in this `else` block, then it means somebody changed it away from the default. This would not detect the case where someone has set it to "true", but that also doesn't seem terribly important to warn people about.

Alternatively, I could refactor `feature_flag.rb` a bit to add a warning if `value` is not nit here: https://github.com/bundler/bundler/blob/66e4215dbf87fedf2fdff3532a0742340c4e02f2/lib/bundler/feature_flag.rb#L22, but that would be a bit of shuffling, and I'm not sure it's worth the effort.

Whatever you think is best!

Reply:
Maybe the easiest (and still correct) thing to do for now is to change the deprecation text to: "Setting the `github.https` to false is deprecated and won't be supported in the future", or something like that?

Reply:
I like that! I've made this change.